libnmstate/state.py

```diff
@@ -32,6 +32,72 @@
 from libnmstate.schema import Interface
 from libnmstate.schema import InterfaceState
 from libnmstate.schema import InterfaceType
+from libnmstate.schema import Route
+
+
+class RouteEntry(object):
+
+    # Mapping dict key name to RouteEntry property name
+    KEY_PROPERTY_MAP = {
```

Comment:
Using this one and `getattr` all over the object is IMO no different from the previous version.
The need for the conversion is limited to the `to_dict`, I do not understand its value in other cases.

BTW, you could have used something like this to report the dict:
`return {k.replace('_', '-'): v for k, v in vars(self).items}`